apps/publish-pipeline-result/src/EventHubClient.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License in the project root for license information.
+
+import { EventHubProducerClient, CreateBatchOptions } from "@azure/event-hubs";
+import { logger } from "./logger";
+
+export class EventHubProducer {
+  private producer: EventHubProducerClient;
+  constructor(sasUrl: string) {
+    this.producer = new EventHubProducerClient(sasUrl);
+  }
+
+  public async send(eventsToSend: string[], partitionKey?: string) {
```

Comment:
Can you split the function to small part functions?

Reply:
done